jamjar/jamjar/tasks/transcode_video.py

```diff
@@ -95,75 +100,61 @@ def delete_source(self):
         video_dir = self.video.get_video_dir()
         shutil.rmtree(video_dir)
 
+    def update_jamstarts(self):
+        concert = self.video.concert
+
+        relevant_edges_query = Q(confidence__gt=settings.CONFIDENCE_THRESHOLD) & Q(video1__concert_id=concert.id) & Q(video2__concert_id=concert.id)
+        relevant_edges = Edge.objects.filter(relevant_edges_query)
+
+        # unique the resulting edges and make all edges positive
+        digraph_edges = []
+        seen_video_ids = set()
+        for edge in relevant_edges:
+            video_ids = (edge.video1_id, edge.video2_id)
+
+            if video_ids in seen_video_ids or video_ids[0] == video_ids[1]:
+                continue
+            else:
+                seen_video_ids.add(video_ids)
+
+            # if the edge is negative, flip it! We only want positive edges
+            if edge.offset < 0:
+                edge.offset *= -1
+                edge.video1_id, edge.video2_id = edge.video2_id, edge.video1_id
+
+            edge_data = (edge.video2_id, edge.video1_id, edge.offset)
+            digraph_edges.append(edge_data)
+
+        # build a graph from the resulting edges
+        digraph = nx.DiGraph()
+        digraph.add_weighted_edges_from(digraph_edges)
+
+        # find all videos in the subgraph
+        video_ids_in_subgraph = nx.shortest_path(digraph.to_undirected(), self.video.id).keys()
```

Comment:
That seems like it would be the more straightforward way to do it.  If you don't wanna do it now that's fine, but you may wanna throw a comment in there about it.